atmcfg/atmcfg_test.go

```diff
@@ -140,6 +160,46 @@ func TestAddUser(t *testing.T) {
 	}
 }
 
+func TestAddIndexConfig(t *testing.T) {
+	config := automationConfigWithIndexConfig()
+	index := &opsmngr.IndexConfigs{
+		DBName:         "test1",
+		CollectionName: "test2",
+		RSName:         "test2",
+		Key: [][]string{
+			{
+				"test", "test",
+			},
+		},
+		Options:   nil,
+		Collation: nil,
+	}
+
+	var err error
+
+	t.Run("AutomationConfig not initialized", func(t *testing.T) {
+		err = AddIndexConfig(nil, index)
+		if err == nil {
+			t.Error("AddIndexConfig did not throw an exception")
+		}
+	})
+
```

Comment:
In these two comments https://github.com/mongodb/go-client-mongodb-ops-manager/pull/20#discussion_r411542009 and 
https://github.com/mongodb/go-client-mongodb-ops-manager/pull/20#discussion_r411551296 I understood that you were asking for the scenario with an automationCongif = nil even if you have specifically asked for the `indexConfig`.  The reason for that is that I have confused the configIndex with the automationConfig . Sorry about that 😞

Reply:
Do you want me to remove the scenario with the `automationConfig = nil`?

Reply:
For the comments that's ok we're working on it, I may suggest one additional change here to avoid more confusion, right now we have the filed `IndexConfigs` inside the AutomationConfig and the type `IndexConfigs` let's rename the type to `IndexConfig` singular, the property makes sense to keep plural as is a collection but this separation can help us avoid other confusions

For your extra scenario you have a validation for `automationConfig` to not be nil so that scenario should stand to improve our coverage

try running `go tool cover -html=coverage.out` after `make test` to check out our current test coverage